tests/conftest.py

```diff
@@ -234,6 +234,11 @@ def pip_with_index_conf(make_pip_conf):
     )
 
 
+@pytest.fixture(autouse=True)
+def pip_with_empty_conf(make_pip_conf):
```

Comment:
This was added because many unit tests make assumptions about default pip.conf. Specifically if you run unit tests on master with a pip.conf that has an extra-index-url the url will get added to output of many tests causing them to fail. This makes all tests that don't specify pip.conf file use empty one and avoids needing to modify your global pip.conf for tests.